Python/chapter03/p02_stack_min/miguelHx.py

```diff
@@ -0,0 +1,248 @@
+"""
+Python version 3.7.0
+3.2 - Stack Min
+How would you design a stack which, in addition to
+push and pop, has a function min which returns the minimum element?
+Push, pop and min should all operate in O(1) time.
+"""
+
+import copy
+import unittest
+import sys
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Generic, TypeVar
+from typing import List, Optional, Iterator
+
+T = TypeVar('T')
+C = TypeVar('C', bound='Comparable')
+
+class Comparable(Generic[T, C]):
+    @abstractmethod
+    def __lt__(self, other: T):
+        pass
```

Comment:
```suggestion
from typing import Protocol

T = TypeVar('T', bound='Comparable')

class Comparable(Protocol):
    @abstractmethod
    def __lt__(self, other: T) -> bool:
        pass
```

from https://github.com/python/typing/issues/59#issuecomment-353878355

`C` isn't used as a type anywhere, so it can go.

Reply:
Committed suggested change here: 3c8ceb5a7249732782b5828321325233cb062008
Removed remaining C usage here:  6d9c43e99e208cde9e734e24591a721c5b2d5c5a